andela/Ah-backend-xmen
# #162725274 Highlight and comment on any text in an article

### What does this PR do?
This PR enables users to highlight and comment on any text in an article. 

#### Description of Task to be completed?
When a user is scrolling through an article and wants to comment on a particular part, he/she is able to select the desired text and comment about it. 

#### How should this be manually tested?
- Fetch this branch in your terminal and check out into it using `git fetch origin ft-highlight-and-comment-on-article-162725274 && git checkout ft-highlight-and-comment-on-article-162725274`
- To avoid conflicts; in the psql shell, drop the existing database using `drop database <database name>;` Then, create the database using `create database <database name>;` 
- Apply migrations by running `python manage.py migrate comments`
- Start the application by running the server with `python manage.py runserver`
- To highlight a particular text in an article for commenting, the request body will contain the start and ending index positions as well as the comment, i.e. `{
                                                   "body": "a comment body", 
                                                    "highlight_start": <integer value for the beginning of highlighted text>, 
                                                    "highlight_end": <integer value for the end of highlighted text> 
                                                }`
#### Screenshots
<img width="1135" alt="highlight" src="https://user-images.githubusercontent.com/31903212/51846081-f1ddff80-2329-11e9-801f-22505e1fe377.png">

#### What are the relevant pivotal tracker stories?
[#162725274](https://www.pivotaltracker.com/story/show/162725274)

#### Postman Collection
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/77211431fb008ed18e9d)
